clients/src/main/java/org/apache/kafka/common/config/provider/FileConfigProvider.java

```diff
@@ -40,7 +42,13 @@ public class FileConfigProvider implements ConfigProvider {
 
     private static final Logger log = LoggerFactory.getLogger(FileConfigProvider.class);
 
+    public static final String ALLOWED_PATHS_CONFIG = "allowed.paths";
+    public static final String ALLOWED_PATHS_DOC = "A comma separated list of paths that this config provider is " +
+            "allowed to access. If not set, all paths are allowed.";
+    private AllowedPaths allowedPaths = null;
```

Comment:
If someone does not call configure() before get() (as the FileConfigProvider tests previously did) then this will cause an NPE. I think either that should preserve the old behavior (allowing all paths) or should throw an IllegalStateException. There might be an opportunity to attack this class by preventing configure() from being called somehow, so perhaps the IllegalStateException is more secure.

Reply:
This is a good point! As suggested in the comments below, we can make the constructor take in a String for the configValue and set the allowList. And here we can set it to `new AllowedPaths(null)` so that we avoid NPE and it always defaults to the previous behaviour allowing all paths. 

Could there be users who don't call configure first necessarily? I'm worried that throwing IllegalStateException could cause backward compatibility issue.

Reply:
@mimaison WDYT?

Reply:
There is something else to consider if we go down the route of throwing an exception(either NPE or IllegalStateException) when configure() is not called first.  We end up breaking tests for AbstractConfig, which is used to resolve the config variables, by instantiating ConfigProviders with the given configProvider configs and then calling get() method (https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java#L536). We would need to make changes to this class, possibly call configure() first when instantiating the CPs which may or may not have further impact somewhere else. 

Considering this, I'm not sure what's more valuable between defaulting to the previous behaviour of no-op and introducing an exception. What do you both think?

Reply:
> We end up breaking tests for AbstractConfig

@tinaselenge Can you clarify what you mean here?

I think the AbstractConfig unconditionally calls configure() here immediately after instantiation: https://github.com/apache/kafka/blob/24a79c2613e32b1d1615f6f34d63225259740d0e/clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java#L593-L594

Reply:
@gharris1727 you are right. The mock classes extending the provider class needed to be updated as they didn't call `configure()` causing test failures.